staging/kos/pkg/controllers/ipam/ipam.go

```diff
@@ -465,11 +469,34 @@ func (ctlr *IPAMController) processNetworkAttachment(ns, name string) error {
 		return nil
 	} else {
 		lockForStatus, ipForStatus, err = ctlr.pickAndLockAddress(ns, name, att, subnetName, desiredVNI, desiredBaseU, desiredLastU)
-		if err != nil {
+		if isFullSubnetErr(err) && !fullSubnetMsgFound(att.Status.Errors.IPAM) {
+			fullSubnetErr = err
+			naStatusErrs = []string{fullSubnetStatusMsg}
+		} else if err != nil {
 			return err
 		}
 	}
-	return ctlr.updateNAStatus(ns, name, att, nadat, naStatusErrs, subnetRV, lockForStatus, ipForStatus)
+	err = ctlr.updateNAStatus(ns, name, att, nadat, naStatusErrs, subnetRV, lockForStatus, ipForStatus)
+	if fullSubnetErr != nil {
+		if err != nil {
+			return fmt.Errorf("%s. %s", fullSubnetErr.Error(), err.Error())
```

Comment:
s/%s. %s/%s; and %s/

Typically the `Error()` of an `error` is an English clause (which is like a sentence but does not have initial capital nor final punctuation and is suitable for composing into a larger clause).

Reply:
I fixed this locally, I'm waiting to push it until we reach consensus on the other issues you brought up to avoid multiple tiny commits in case we chose to change other things.

Reply:
Fixed in c2bc019.